open-company/open-company-web
# Code cleanup

Card: https://trello.com/c/vgDv1Mak

### Components

Removed components:
```
- comment-reactions
- comments
- home
- mobile-boards-list
- org-editor
- ui/icon
- ui/media-attachments
```

Other components:
- interactions-summary: renamed to comments-summary, before it included comments-summary and reactions-summary and a component that used both called interactions-summary. Now we need only the first so i renamed the namespace and removed the other 2.


### Namespaces and fns

#### Removed namespaces:

```
- oc.web.lib.medium-editor-exts
- oc.web.lib.oc-colors
- test.test.web.lib.utils
```

#### Cleaned namespaces removing not used functions:

```
- oc.web.lib.responsive
- oc.web.lib.utils
- oc.web.api
```

removing some functions from `oc.web.lib.utils`, the utils tests file was empty so i removed it too:
```
test.oc.web.lib.utils
```

#### Removed all not used images.

#### Removed all not used deps

```
cljsjs/emojione-picker
```

#### Removed not used js and css assets

```
resources/public/css/emojione.sprites.css
resources/public/lib/GoogleSheets/*
resources/public/lib/Sortable.js/*
resources/public/lib/autogrow/*
resources/public/lib/clamp/*
resources/public/lib/hallo/hallo.js
resources/public/lib/jquery-ui/*
resources/public/lib/js-utils/svg-utils.js
resources/public/lib/select2/*
resources/public/lib/tooltip/tooltip.js
resources/public/lib/truncate/jquery.truncate.js
```

### Tests

During this test check for any missing asset image or any weird 404 in the console. Usually we have a `main.css` 404 when loading the marketing site on staging/beta and a 404 if the whats-new board is set but not present (staging is 404ing right now, beta is not).

#### Test marketing site

- [x] home page is ok? Good
- [x] navigate to /about, all good? Good
- [x] navigate to /slack, all good? Good
- [x] navigate to /pricing, all good? Good
- [x] navigate to /privacy, all good? Good
- [x] navigate to /terms, all good? Good
- [x] repeat the above with a mobile device or with chrome with device mode ON. All good? Good

#### Testing desktop app

- [x] visit /login
- dismiss the login overlay
- [x] navigate to about, slack and pricing using the links in the footer, all good? Good
- click on Start in top right
- [x] sign up
- [x] compose
- [x] use medium editor formatting
- [x] add an emoji to the headline with the emoji picker at the bottom
- [x] add an emoji to the headline with the autocomplete
- [x] add an emoji to the body with the emoji picker at the bottom
- [x] add an emoji to the body with the autocomplete
- [x] add a video in the body
- [x] add an image in the body
- [x] add attachments
- [x] save as draft
- go to draft
- click on the post
- [x] edit some more
- [x] post the draft
- [x] compose again
- [x] post an update with some content (whatever you like)
- [x] go to AP
- [x] switch btw grid and stream view
- [x] go to a board
- [x] switch btw grid and stream view
- [x] share a post to email
- [x] edit a post
- [x] save edited post
- click on the user menu
- [x] edit the org data and save
- [x] check the invite people tab
- [x] invite 2 or 3 email users with different roles
- [x] go to team management and check the users
- [x] change 2 users role and remove one
- [x] go back to the dashboard
- click on the user menu
- [x] click on profile, appeared? Good
- [x] update your name and avatar
- [x] log back in

#### Testing mobile app

- [x] repeat the above steps from mobile (filter out the features we don't expose on mobile like video, image and attachments add)
